content/blog/coreboot-on-talos2.md

```diff
@@ -0,0 +1,307 @@
+---
+title: coreboot and Heads as an alternative firmware for OpenPOWER Talos II
+author: Krystian Hebel
+tags:
+  - openpower
+  - power9
+  - firmware
+  - coreboot
+  - hostboot
+date: 2024-07-09
+draft: true
+---
+
+This blog post presents coreboot ([spelled in lower case characters](https://doc.coreboot.org/#spelling-of-coreboot),
+even when it is the first word in a sentence) and Heads as an alternative to
+Hostboot and Skiroot/Petitboot, respectively.
+
+As described on the [project's page](https://coreboot.org),
+
+> coreboot is an extended firmware platform that delivers a lightning fast and
+> secure boot experience on modern computers and embedded systems.
+
+{{< image src="blog/coreboot.svg" >}}
+
+It aims to do the bare minimum required to make the hardware usable and pass
+the control to next program, called the payload. In case of Talos II, that
+payload is Skiboot, with [few changes on top](https://github.com/Dasharo/skiboot/tree/raptor-cs_talos-2)
+to make it play along with Heads.
+
+Speaking of [Heads](https://github.com/linuxboot/heads), it is:
+
+> a minimal Linux that (...) provides a secure, flexible boot environment for
+laptops, workstations and servers.
+
+Heads provides a bootloader menu that starts final operating system through
+kexec call. This is very similar to what Petitboot does, but Heads puts
+security above everything else.
+
+It is possible to use coreboot without Heads, but not the other way around.
```

Comment:
I thought heads could be used to replace only petitboot?

Reply:
Does it even start without CBMEM and CBFS?

Reply:
@tlaurion ?

Reply:
> Does it even start without CBMEM and CBFS?

Would have to retest, but I thought petitboot/heads could be used interchangeably through openbmc virtual flash (mboxctl)

Reply:
But I guess not. Docs at https://docs.dasharo.com/variants/talos_2/initial-deployment/ is pretty clear: its either hostboot/coreboot and therefore if coreboot, petitboot/heads. Sorry for the noise, I don't see need of testing following hypothesis invalidation by thought experiment only.

Reply:
Ok, closing this thread then